src/Foundatio/Utility/OptionsBuilder.cs

```diff
@@ -1,17 +1,28 @@
 ﻿using System;
 
 namespace Foundatio {
-    public interface IOptionsBuilder<out T> {
-        T Target { get; }
+    public interface IOptionsBuilder {
+        object Target { get; }
+    }
+
+    public interface IOptionsBuilder<T> : IOptionsBuilder {
+        T Build();
+    }
+
+    public static class OptionsBuilderExtensions {
+        public static T Target<T>(this IOptionsBuilder builder) {
+            return (T)builder.Target;
+        }
     }
 
     public class OptionsBuilder<T> : IOptionsBuilder<T> where T : class, new() {
         public T Target { get; } = new T();
+        object IOptionsBuilder.Target => Target;
 
-        public static T Build(Action<OptionsBuilder<T>> config) {
-            var builder = new OptionsBuilder<T>();
-            config?.Invoke(builder);
-            return builder.Target;
+        public virtual T Build() {
+            return Target;
         }
     }
+
+    public delegate TBuilder Builder<TBuilder, TOptions>(TBuilder builder) where TBuilder: class, IOptionsBuilder<TOptions>, new();
```

Comment:
The delegate name should not so simple that it's too easily confused with mscorlib classes.

Reply:
It has a generic constraint that makes it so that its not useable with random things. I wanted a name that was simple since we have to pass in both type parameters to it.